libethereum/Client.cpp

```diff
@@ -290,7 +290,10 @@ void Client::reopenChain(ChainParams const& _p, WithExisting _we)
     }
 
     if (auto h = m_host.lock())
+    {
         h->reset();
+        h->completeSync();  // set sync state to idle for mining
```

Comment:
@gumb0 this looks like Kostil` around m_state variable that must be set to Idle.  but I am afraid that changing it's in internal funcs might break the client.

Reply:
You move it down the stack instead of up the stack, as I asked. This is probably even worse.

Reply:
Where is `setChainParams` called for your case? Can you put `completeSync` call there?

Reply:
setChainParams is test client method. so the best place is to keep it where it was. or make it a separate method of the test client. (which is unnecessary)

Reply:
There is already `ClientTest::completeSync()`, you could call it from from RPC method implementation `Test::test_setChainParams()` (add `asClientTest(m_eth).completeSync();` there), similar to what happens in `Test::test_rewindToBlock()`. It still would be ugly, but less so.

Starting `eth --test` doesn't enable sync by itself, can you find out at which moment does it become enabled? (the stack where `BlockChainSync::m_state` changes)

Reply:
the moment where it changes is strange. it works for 1 time and then it set to notSynced after blockchain reset.